frontend/src/lib/components/PropertyFilters/components/FilterRow.tsx

```diff
@@ -60,18 +74,19 @@ export const FilterRow = React.memo(function FilterRow({
             {disablePopover ? (
                 <>
                     {filterComponent(() => setOpen(false))}
-                    {!!Object.keys(filters[index]).length && (
-                        <CloseButton
-                            onClick={() => onRemove(index)}
-                            style={{
-                                cursor: 'pointer',
-                                float: 'none',
-                                paddingLeft: 8,
-                                alignSelf: 'flex-start',
-                                paddingTop: 4,
-                            }}
-                        />
-                    )}
+                    {!!Object.keys(filters[index]).length &&
+                        !isFirstMatchingPinnedFilter(pinnedFilters, filters, index, filters[index]) && (
```

Comment:
The result of `isFirstMatchingPinnedFilter(pinnedFilters, filters, index, filters[index])` seems to be called often. What about promoting it to a local variable `isPinned` instead?

Also, it doesn't seem to do anything. The pinned filters themselves are just being shown as text and looped over in a different component...?